src/core/evaluator.js

```diff
@@ -2407,14 +2469,16 @@ class PartialEvaluator {
               self.ensureStateFont(stateManager.state);
               continue;
             }
-            var items = args[0];
-            var offset;
-            for (var j = 0, jj = items.length; j < jj; j++) {
-              if (typeof items[j] === "string") {
-                buildTextContentItem(items[j]);
-              } else if (isNum(items[j])) {
-                ensureTextContentItem();
 
+            let lastAdvance = 0;
+            const spaceFactor =
+              ((textState.font.vertical ? 1 : -1) * textState.fontSize) / 1000;
+            const elements = args[0];
+            for (let i = 0, ii = elements.length; i < ii - 1; i++) {
+              const item = elements[i];
+              if (typeof item === "string") {
+                showSpacedTextBuffer.push(item);
+              } else if (isNum(item) && item !== 0) {
```

Comment:
Nit: This is obviously pre-existing code, but I can't help wonder if replacing the `isNum` call with an inline `typeof time === "number"` instead wouldn't be a tiny bit more efficient!?